File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    vector<char> remove;
    for(int i=0;i<number.size();i++)
    {
        int num=number[i]-'0';
        bool possible=true;
        for(int j=i+1;j<number.size();j++)
        {
            if(k==0)
                break;
            
            if(num<number[j]-'0')
            {
                if(k>=j-i)
                {
                    possible=false;
                    k-=j-i;
                    i=j-1;
                }
                
                break;
            }
        }
        
        if(k==1 && i==number.size()-1)
            break;
            
        if(possible)
        {
            answer+=number[i];
        }
        
    }
    
    return answer;
}

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public string solution(string number, int k) {
        string answer = "";
        var st = new Stack<int>();
        foreach(var c in number)
        {
            int num = c-'0';
            
            if(st.Count==0)
            {
                st.Push(num);
            }
            else if(st.Peek()<num && k!=0)
            {
                while(st.Peek()<num && k!=0)
                {
                    st.Pop();
                    k--;
                    
                    if(st.Count==0)
                        break;
                }
                st.Push(num);
            }
            else
            {
                st.Push(num);
            }
        }
        
        while(k>0)
        {
            st.Pop();
            k--;
        }
        
        char[] answers = new char[st.Count];
        int count = st.Count;
        for (int i = count - 1; i >= 0; i--) 
        {
            answers[i] = (char)(st.Pop() + '0');
        }

        return new string(answers);
    }
}